packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2509,30 +2510,78 @@ function warnIfNotCurrentlyActingUpdatesInDEV(fiber: Fiber): void {
 export const warnIfNotCurrentlyActingUpdatesInDev = warnIfNotCurrentlyActingUpdatesInDEV;
 
 let componentsWithSuspendedDiscreteUpdates = null;
+let componentsThatCallSuspendedDiscreteUpdates = new Set();
```

Comment:
This should be initialized to `null` so that it doesn't allocate memory on start up. Same as `componentsWithSuspendedDiscreteUpdates`.

Also let's rename these to remove the "discrete" since that no longer applies. Suggested names: `componentsThatTriggeredHighPriSuspend` and `componentsThatSuspendedAtHighPri`

Reply:
This set also needs to be reset inside `prepareFreshStack`. Look for all the places that `componentsWithSuspendedDiscreteUpdates` is referenced and do the same thing.